cmd/containerd-shim-runhcs-v1/clone.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"context"
+
+	"github.com/Microsoft/hcsshim/internal/clone"
+	"github.com/Microsoft/hcsshim/internal/uvm"
+)
+
+// saveAsTemplate saves the UVM and container inside it as a template and also stores
+// the relevant information in the registry so that clones can be created from this
+// template.
+// Saving is done in following steps:
+// - First remove all the NICs associated with the host.
+// - Close the GCS connection.
+// - Save the information about the templtae that will be needed during cloning
```

Comment:
small nit: *template

Reply:
another small nit: I'm not sure the steps in the description are that necessary, the code reads pretty clearly. Maybe it would be more helpful to mention why some of these things need to be done instead?